pkg/monitortests/network/onpremkeepalived/keepalived_priority_check.go

```diff
@@ -0,0 +1,265 @@
+		vipType = "INGRESS_1"
+	}
+	switch {
+	case priorityRe.MatchString(logLine.Line):
+		subMatches := priorityRe.FindStringSubmatch(logLine.Line)
+		subNames := priorityRe.SubexpNames()
+		previousPriority := ""
+		newPriority := ""
+		for i, name := range subNames {
+			switch name {
+			case "PREV_PRIO":
+				previousPriority = subMatches[i]
+			case "CURR_PRIO":
+				newPriority = subMatches[i]
+			}
+		}
+		g.recorder.AddIntervals(
+			monitorapi.NewInterval(monitorapi.SourcePodLog, monitorapi.Info).
+				Locator(logLine.Locator).
+				Message(monitorapi.NewMessage().
+					Reason(monitorapi.OnPremLBPriorityChange).
+					WithAnnotation(monitorapi.AnnotationPreviousPriority, previousPriority).
+					WithAnnotation(monitorapi.AnnotationPriority, newPriority).
+					HumanMessage(logLine.Line),
+				).
+				Build(logLine.Instant, logLine.Instant),
+		)
+	case masterRe.MatchString(logLine.Line):
+		g.recorder.AddIntervals(
+			monitorapi.NewInterval(monitorapi.SourcePodLog, monitorapi.Info).
+				Locator(logLine.Locator).
+				Message(monitorapi.NewMessage().
+					Reason(monitorapi.OnPremLBTookVIP).
+					WithAnnotation(monitorapi.AnnotationVIP, vipType).
+					HumanMessage(logLine.Line),
+				).
+				Build(logLine.Instant, logLine.Instant),
+		)
+	case backupRe.MatchString(logLine.Line):
+		g.recorder.AddIntervals(
+			monitorapi.NewInterval(monitorapi.SourcePodLog, monitorapi.Info).
+				Locator(logLine.Locator).
+				Message(monitorapi.NewMessage().
+					Reason(monitorapi.OnPremLBLostVIP).
+					WithAnnotation(monitorapi.AnnotationVIP, vipType).
+					HumanMessage(logLine.Line),
+				).
+				Build(logLine.Instant, logLine.Instant),
+		)
+	}
+
+}
+
+func (*operatorLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	priorityIntervals := finalIntervals.Filter(func(eventInterval monitorapi.Interval) bool {
+		if eventInterval.Message.Reason == monitorapi.OnPremLBPriorityChange {
+			return true
+		}
+		return false
+	})
+	testName := fmt.Sprintf("[Jira:\"Networking / On-Prem Load Balancer\"] on-prem loadbalancer must achieve full priority")
+
+	neededPriority := "65"
+	achievedPriority := false
+	for _, interval := range priorityIntervals {
+		if interval.Message.Annotations[monitorapi.AnnotationPriority] == neededPriority {
+			achievedPriority = true
+		}
+	}
+
+	ret := []*junitapi.JUnitTestCase{}
+	if achievedPriority {
+		ret = append(ret, &junitapi.JUnitTestCase{
+			Name: testName,
+		})
+	} else {
+		ret = append(ret,
+			&junitapi.JUnitTestCase{
+				Name: testName,
+				FailureOutput: &junitapi.FailureOutput{
+					Message: fmt.Sprintf("no master achieved priority %s", neededPriority),
+					Output:  fmt.Sprintf("no master achieved priority %s", neededPriority),
+				},
+			},
+		)
+	}
+	// Force the test to flake even if it failed
+	ret = append(ret, &junitapi.JUnitTestCase{
+		Name: testName,
+	})
```

Comment:
From the new test detection: 

"[Jira:"Networking / On-Prem Load Balancer"] on-prem loadbalancer must achieve full priority" [Total: 36, Pass: 36, Fail: 0, Flake: 30]

Anything that can be done about that flake rate? Is that expected at this point? Actually now that I think of it, does this monitortest need to disable itself if it's not on an on-prem platform? I don't see anything in here preventing this from running on the normal clouds.